### PR TITLE
SystemDS Documentation

### PR DESCRIPTION
**Pull request by Baunsgaard**
Brute change of documentation page to reflect name change.

To fully enforce this two more steps are needed:

1. https://github.com/apache/systemml-website
2. Access to SystemML website server on https://systemml.apache.org/ 

Furthermore, we might want to consider starting fresh in the documentation.

### CODE REVIEW

**Comment by j143** [comment]
Hi @Baunsgaard,  priliminary comments:

- [ ] Documentation _name refactoring_ can be done on feature by feature basis.

  i.e.,
        > 1. changing all the `MLLearn` related naming at once
        > 2. and going for `MLContext` related in the next step
  this would help us get through this more sanely. or if you have confidence we would still go with the present changes. :)
-  [ ] Also, renaming of the following files can be considered (exhaustive list)
1. [developer-tools-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-5a20cd23dee2cf3706698331acae778e)
2. [contributing-to-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-7c60d52419aa7274236adc2abd084484)
3. [alg-ref/SystemML_Algorithms_Reference.tex](https://github.com/apache/systemml/pull/877/files#diff-7a90e6afce66bf963deeb423d0db9a6c)
4. [developer-tools-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-5a20cd23dee2cf3706698331acae778e)

Thank you.

File: README.md [hunk 1]
```diff
@@ -17,26 +17,49 @@ limitations under the License.
...
-    $ brew install python
-    $ pip install Pygments
-    $ gem install pygments.rb
+4. Launch the Documentation locally
+
+   - `jekyll serve -w`
+   - This is done from the root of the gh-pages branch of the system.
+   - The serving will per default be on localhost port 4000 [Link](http://localhost:4000)
+
+## Mac Install (Depricated)
```
Comment (j143) [inline README.md:44]: _Deprecated_ is more popular though.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+## VS Code
+
+.settings/
+.vscode/
+
+## Inproper copy when switching branch
+
+target/
+src/
```
Comment (j143) [inline .gitignore:15]: Hi, can this _.gitignore_ file be moved in another PR?.

Comment (Baunsgaard) [reply .gitignore:15]: https://github.com/apache/systemml/pull/879

Comment (Baunsgaard) [reply .gitignore]: Also removed it from this pull request.

File: README.md (hunk 1 above)
Comment (Baunsgaard) [reply README.md]: Done

**Comment by Baunsgaard** [comment]
When switching the MLContext to DSContext it does not allign with the new python.
These again, have to be rewritten.

**Comment by j143** [comment]
Hey!
> When switching the MLContext to DSContext it does not allign with the new python.
> These again, have to be rewritten.

it seems `MLContext` is not changed in SystemDS. So, the last two commits can be reverted
1. [5dc3528](https://github.com/apache/systemml/pull/877/commits/5dc35285695d70fa161078d804a959546dfd4c37)
2. [639009b](https://github.com/apache/systemml/pull/877/commits/639009b547e4258594f031d8ed7155a208c84647)

And although this PR is good. Let us wait for a day (till 9th of April) more for others to chime in with comments.

**Comment by j143** [comment]
> * [ ]  Also, renaming of the following files can be considered (exhaustive list)
> 
> 1. [developer-tools-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-5a20cd23dee2cf3706698331acae778e)
> 2. [contributing-to-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-7c60d52419aa7274236adc2abd084484)
> 3. [alg-ref/SystemML_Algorithms_Reference.tex](https://github.com/apache/systemml/pull/877/files#diff-7a90e6afce66bf963deeb423d0db9a6c)
> 4. [developer-tools-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-5a20cd23dee2cf3706698331acae778e)

Also can this be done. thanks.

**Comment by Baunsgaard** [comment]
> > * [ ]   Also, renaming of the following files can be considered (exhaustive list)
> > 
> > 
> > 1. [developer-tools-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-5a20cd23dee2cf3706698331acae778e)
> > 2. [contributing-to-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-7c60d52419aa7274236adc2abd084484)
> > 3. [alg-ref/SystemML_Algorithms_Reference.tex](https://github.com/apache/systemml/pull/877/files#diff-7a90e6afce66bf963deeb423d0db9a6c)
> > 4. [developer-tools-systemml.md](https://github.com/apache/systemml/pull/877/files#diff-5a20cd23dee2cf3706698331acae778e)
> 
> Also can this be done. thanks.

1. and 4. is the same. 

and the change in latex is somewhat larger.

**Comment by j143** [comment]
1. can this be `.gitignore` removed, 
2. Does it help opening new pull request for `README.md`? - Yes!
3. So the present pull request can be renamed to **[MINOR] Name Refactor `SystemML` to `SystemDS`**
  a. `_layouts/global.html` 1 file
  b.  `alg-ref/*.tex` - 19 files
               ` .gitignore` - 1 file
                `*.bib` - 1 file
   c. `algorithms-*.md` - 9 files
   d. `beginner-guide-*.md` - 4 files

Feel free to breakdown this task into any (upto 5Nos.) of steps, so that each step can be reviewed crystal clear. thanks.

**Comment by Baunsgaard** [comment]
>     1. can this be `.gitignore` removed,
> 
>     2. Does it help opening new pull request for `README.md`? - Yes!
> 
>     3. So the present pull request can be renamed to **[MINOR] Name Refactor `SystemML` to `SystemDS`**
>        a. `_layouts/global.html` 1 file
>        b.  `alg-ref/*.tex` - 19 files
>        ` .gitignore` - 1 file
>        `*.bib` - 1 file
>        c. `algorithms-*.md` - 9 files
>        d. `beginner-guide-*.md` - 4 files
> 
> 
> Feel free to breakdown this task into any (upto 5Nos.) of steps, so that each step can be reviewed crystal clear. thanks.

1. No, my error should be fixed now.
2. Will do
3. Not sure what you want, but i think it makes the most sense to keep the refactor together.

File: _config.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@ exclude:
   - lang-ref
 
 # These allow the documentation to be updated with newer releases
-SYSTEMML_VERSION: 1.3.0-SNAPSHOT
+SYSTEMDS_VERSION: 2.0.0-SNAPSHOT
```
Comment (j143) [inline _config.yml:18]: I remember seeing `0.3.0-SNAPSHOT` during mvn build, not sure what the decision was.

File: deep-learning.md [hunk 4]
```diff
@@ -29,21 +29,21 @@ limitations under the License.
...
-| Can be invoked using spark-shell                                                                     | Yes. Please see [Scala MLContext API](http://apache.github.io/systemml/spark-mlcontext-programming-guide)  | Limited support                                                                                               | No                                                                              |
+| Can be invoked using pyspark                                                                         | Yes. Please see [Python DSContext API](http://apache.github.io/systemml/spark-mlcontext-programming-guide) | Yes.                                                                                                          | Yes.                                                                            |
```
Comment (j143) [inline deep-learning.md:42]: Can this `MLContext` -> `DLContext` change be reverted, as discussed.

File: _config.yml (hunk 3 above)
Comment (Baunsgaard) [reply _config.yml:18]: This is still up for debate, as discussed in the emails.
In my opinion the 2.0 seems the cleanest, since that wouldn't change previous releases.
